test/SArray.jl

```diff
@@ -139,6 +139,13 @@
         @test_throws Exception m[1] = 1
 
         @test Base.dataids(m) === ()
+
+        @test (@view m[:, :]) === m
+        @test (@view m[:, 1]) === @SArray [11, 12]
+        @test (@view m[SVector{2,Int}(1,2), 1]) === @SArray [11, 12]
+        @test (@view m[SMatrix{2,2,Int}(1,2,3,4)]) === m
+        @test (@view m[SOneTo(2), 1]) === @SArray [11, 12]
+        @test (@view m[1, 1])[] === m[1, 1]
```

Comment:
Could you write a test like `(@view m[1, 1]) isa Scalar`? If `(@view m[1, 1])` returned just `m[1,1]` your test would still pass.

Also, could you wrap left side of each equality test here with `@inferred`?

Reply:
I've updated the tests as suggested, added `@inferred` and checked explicitly for `Scalar` in scalar indexing. I've also added `CartesianIndex` to the list of types that use this view method, as they are analogous to scalar indexing.